### DISCUSSION TITLE
mini.pick: Is there a way to use colored text in the main window or preview window?

### DISCUSSION THREAD

**Discussion by heygarrett** [nvim-mini/mini.nvim#523] [Q&A]
I'm trying to create a picker for commits, and I'd like to use colored text for both the list of commits (to differentiate the hash, relative date, commit message, author email, etc) as well as the diff that would be displayed in the preview window (red for removed lines and green for added lines).

Is there a supported way to do this?

**Comment by heygarrett** [comment]
Well, I figured out that, using the `buf_id` passed to the preview function, I can do:

```lua
vim.bo[buf_id].filetype = "diff"
```

This provides highlighting for the output of `git diff` and `git show`.

**Comment by echasnovski** [comment]
I plan to release `git_commits` picker as a [part of an upcoming 'mini.extra' module](https://github.com/echasnovski/nvim/blob/901f8e693f7834965f8c8fd3aa5fb514a090b74b/lua/mini-dev/extra.lua#L348-L390).

Yes, setting `'diff'` filetype is the way to go to have proper buffer highlighting.

To have highlighting for line parts, update `source.show()` to use `default_show()` followed by setting custom `extmarks` which highlight regions. See `:h extmarks` and `:h nvim_buf_set_extmarks`.
